pkg/eval/platform/platform_test.go

```diff
@@ -7,15 +7,29 @@ import (
 	"github.com/elves/elvish/pkg/eval"
 )
 
+const (
+	testHostname = "mach1.domain.tld"
+	testMachname = "mach1"
+)
+
+func hostnameMock() (string, error) {
+	return testHostname, nil
+}
+
 var That = eval.That
 
 func TestPlatform(t *testing.T) {
-	setup := func(ev *eval.Evaler) { ev.Builtin.AddNs("platform", Ns) }
+	setup := func(ev *eval.Evaler) {
+		ev.Builtin.AddNs("platform", Ns)
+		osHostname = hostnameMock
```

Comment:
This isn't really part of the setup for the `Evaler`, so better put it outside the `setup` function.

Also, it's good practice to reset `osHostname` to its original value, even if that doesn't really matter here. So I'd write the setup as:

```go
func TestPlatform(t *testing.T) {
    osHostname = hostnameMock
    defer func() { osHostname = os.Hostname }
    setup := ...
    ...
}
```

Reply:
Done, although I'll point out it should save the original value in a temp var and restore that in the deferred function rather than hardcoding `os.Hostname`. There is no reason for the unit test to be that tightly coupled to the code being tested.